File: Loenn/entities/strawberry_jam/grab_temple_gate.lua
local drawableSprite = require("structs.drawable_sprite")

local grabTempleGate = {}

grabTempleGate.name = "CommunalHelper/SJ/GrabTempleGate"
grabTempleGate.depth = -9000
grabTempleGate.canResize = {false, false}

grabTempleGate.placements = {
    {
        name = "open",
        data = {
            closed = false,
        }
    },
    {
        name = "closed",
        data = {
            closed = true,
        }
    }
}

local texture = "objects/CommunalHelper/strawberryJam/grabTempleGate/TempleDoor05"

function grabTempleGate.sprite(room, entity)
    local sprite = drawableSprite.fromTexture(texture, entity)

    if not entity.closed then
        sprite.color = {1.0, 1.0, 1.0, 0.45}
    end

    sprite:setJustification(0.5, 0.0)
    sprite:addPosition(4, 0)

    return sprite
end

return grabTempleGate

File: Ahorn/entities/strawberryJam/GrabTempleGate.jl
module CommunalHelperSJGrabTempleGate

using ..Ahorn, Maple

@mapdef Entity "CommunalHelper/SJ/GrabTempleGate" GrabTempleGate(x::Integer, y::Integer, closed::Bool = false)

const placements = Ahorn.PlacementDict(
    "Grab Temple Gate (Open) (Strawberry Jam) (Communal Helper)" => Ahorn.EntityPlacement(
        GrabTempleGate,
        "point",
        Dict{String, Any}(
            "closed" => false
        )
    ),
    "Grab Temple Gate (Closed) (Strawberry Jam) (Communal Helper)" => Ahorn.EntityPlacement(
        GrabTempleGate,
        "point",
        Dict{String, Any}(
            "closed" => true
        )
    )
)

const texture = "objects/CommunalHelper/strawberryJam/grabTempleGate/TempleDoor05";

function Ahorn.selection(entity::GrabTempleGate)
    x, y = Ahorn.position(entity)
    return Ahorn.Rectangle(x - 3, y, 15, 48)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::GrabTempleGate, room::Maple.Room)
    Ahorn.drawImage(ctx, texture, -8, 0)
end

end